src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.CompleteCommand;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.task.TaskType;
+
+/**
+ * Encapsulates the parsing of user inputs.
+ */
+public class Parser {
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_FIND = "find";
+    private static final String COMMAND_COMPLETE = "done";
+    private static final String COMMAND_ADD_TODO = "todo";
+    private static final String COMMAND_ADD_EVENT = "event";
+    private static final String COMMAND_ADD_DEADLINE = "deadline";
+    private static final String COMMAND_DELETE = "delete";
+
+    /**
+     * Processes the full user input.
+     *
+     * @param fullCommand user's input
+     * @return one of the subclasses of Command that represents the application logic
+     * @throws DukeUnknownCommandException if the user's command is unknown
+     */
+    public static Command parse(String fullCommand) throws DukeUnknownCommandException {
+        StringBuilder commandInput = new StringBuilder();
+        StringBuilder argsInput = new StringBuilder();
+        boolean commandFound = false;
```

Comment:
Maybe the boolean variable can be renamed to sound more like a boolean, for example, name it to "isCommandFound"?

Reply:
Thanks for pointing these out 😄 I've changed them. 

Also, I've changed some relevant ones (previously `Task.decodeCompletionFlag()` and `Task.descriptionContains`).